### ISSUE TITLE
Bug with new commit from Oct 29, 2023

### ISSUE THREAD

**Issue by dpieterse** [Bill-Gray/lunar#23]
After the commit you made on Oct 29 (86a705622f2e5d8a2907f9f00093e74dc6f1d7d8), integrat no longer works for me. This is because of the lines: _assert( got_it);_ and _assert( position_cache);_ where the assertion fails.

I run integrat on the MPC Orbital Database (MPCORB.DAT) via:
`integrat MPCORB_version20231126T1500.DAT MPCORB_integrated.DAT 2460276.5 -f linux_m13000p17000.441`

**Comment by Bill-Gray** [comment]
Hmmm... `integrat` hasn't worked with the MPCORB.DAT format since commit 2b888b4fce77,  on 2022 Oct 25,  a little over a year ago.  There were several reasons for this,  but the really key one was that I had to handle comets,  and they sometimes have eccentricities greater than one.  In particular,  we had situations where an MPCORB.DAT comet with e barely under 1 integrated to have e barely greater than 1.

I switched to use of the ['.sof' ("standard" orbit format)](https://www.projectpluto.com/orb_form.htm),  which meant _all_ orbits,  not just elliptical ones,  could be handled.  To do the conversion,  run the `mpc2sof` code in this repository.

I do recognize that there's a problem here if you have code that ingests only the MPCORB.DAT format.  Ideally,  we'd have something better for a format that would allow exchange between programs.  I obviously hoped that `.sof` would do that.  But it hasn't caught on;  we continue to have a bazillion mutually incompatible orbital element formats.
